internal/scrapertest/compare.go

```diff
@@ -22,9 +22,18 @@ import (
 	"go.uber.org/multierr"
 )
 
+// CompareOption is applied by the CompareMetricSlices function
+// to mutates an expected and/or actual result before comparing
+type CompareOption func(expected, actual pdata.MetricSlice)
+
 // CompareMetricSlices compares each part of two given MetricSlices and returns
 // an error if they don't match. The error describes what didn't match.
-func CompareMetricSlices(expected, actual pdata.MetricSlice, checkValues bool) error {
+func CompareMetricSlices(expected, actual pdata.MetricSlice, options ...CompareOption) error {
+
+	for _, option := range options {
+		option(expected, actual)
+	}
```

Comment:
Should we "clone" the expected/actual values before mutating them? Just to not have side effects?

Reply:
Probably not necessary most of the time since these values will just be thrown away after being compared, but yeah, we'll save someone some trouble.

Reply:
These are now being cloned.